hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -92,4 +113,49 @@ private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEng
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieEngineContext context,
+                                          HoodieStorage hoodieStorage,
+                                          HoodieTableMetaClient metaClient,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    // Fetch metadata partition paths.
+    List<String> metadataPartitions = FSUtils.getAllPartitionPaths(
+        context,
+        hoodieStorage,
+        metaClient.getBasePath(),
```

Comment:
The passed-in meta client is the data table meta client.  Should the metadata table meta client be constructed and passed in here?

Reply:
I checked the implementation, the metadata table base path was constructed multiple layers inside. No metadata table meta client is needed.

Reply:
Not following your comment.  `FSUtils.getAllPartitionPaths` returns the partitions of the table based on the meta client and base path passed in.  If the passed-in meta client is the data table meta client, the `metadataPartitions` contains the data table partitions.

Reply:
yes it should be metadata base path. You can use `HoodieTableMetadata.getMetadataTableBasePath(metaClient.getBaseBathV2().toString())`

Reply:
fixed.